Fhi.HelseId/Api/Authorization/CustomAuthenticationHandler.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+
+namespace Fhi.HelseId.Api.Authorization
+{
+    /// <summary>
+    /// This is taken from Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler, the only thing we to is to add DPoP for authorazation
+    /// </summary>
+    public class CustomAuthenticationHandler(
+        IOptionsMonitor<JwtBearerOptions> options,
+        ILoggerFactory logger,
+        UrlEncoder encoder,
+        ISystemClock clock) : JwtBearerHandler(options, logger, encoder, clock)
+    {
+        private OpenIdConnectConfiguration? _configuration;
+
+        /// <summary>
+        /// The only change is to add authorization.StartsWith("DPoP ")
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
```

Comment:
❌ - While I don't want to be the one demanding change in the codebase, I feel that this method is too nested and long. If there's some context which I'm not aware of, please let me know.

As it stands, I believe it should be decomposed into smaller, more readable methods.

Reply:
I don't want to touch the authentication method from microsoft as little as possible.

Reply:
Makes sense. Do they have some tests we could take as well? In case it turns out that we have to change it (for whatever reason)?

Reply:
Just my thought: we should NOT do any more changes to this class. The change present is simply to allow dpop type-word to be treated as bearer.

Our proper implementation of dpop validation should be our own fully testes code, and not a modification of this one.

Reply:
I agree @martsve.

There might be some context I don't have: Is this a temporary "fix" to quickly get some dpop support, and this class will be replaced in the future?